src/hb-ot-name-table.hh

```diff
@@ -477,6 +563,38 @@ struct name
     hb_vector_t<hb_ot_name_entry_t> names;
   };
 
+  private:
+  // sometimes NameRecords are not sorted in the font file, so use linear search
+  // here
+  bool has_name_record_with_ids (const hb_ot_name_record_ids_t& record_ids) const
+  {
+    for (const auto& record : nameRecordZ.as_array (count))
+    {
+      if (record.platformID == record_ids.platform_id &&
+          record.encodingID == record_ids.encoding_id &&
+          record.languageID == record_ids.language_id &&
+          record.nameID == record_ids.name_id)
+        return true;
+    }
+    return false;
+  }
+
+  bool all_ascii_characters (hb_bytes_t bytes) const
```

Comment:
I would move this check into hb-subset-input.cc when the entry is first being added to the override map. Then if the string is not valid you can skip adding it to the map and return false from that method to signal to the caller it was not accepted.

Reply:
The problem is hb_utf8_t is not visible to hb-subset-input.cc and `hb-utf.hh` can not be included in `hb-subset-input.hh` because there's circular dependency issue. Any suggestions?

Reply:
Oh my mistake, I can include it in `.cc` not `.hh` file. I'll fix it.

Reply:
Done.